### PR TITLE
Removed legacy ARRAKIS tiles

### PR DESCRIPTION
**Pull request by Mailaender**
Split from https://github.com/OpenRA/OpenRA/pull/4496 as suggested in https://github.com/OpenRA/OpenRA/pull/7714.

### CODE REVIEW

File: OpenRA.Game/Map/TileSet.cs [hunk 1]
```diff
@@ -246,6 +246,9 @@ public byte GetTerrainIndex(string type)
 
 		public byte GetTerrainIndex(TerrainTile r)
 		{
+			if (!Templates.ContainsKey(r.Type))
```
Comment (pchote) [inline OpenRA.Game/Map/TileSet.cs]: You can save a dictionary lookup by using TryGetValue.  See e.g. 965447a42c92aecdbeacef69be4f72a1478ef08b.

Comment (pchote) [inline OpenRA.Game/Map/TileSet.cs]: You may also need 965447a42c92aecdbeacef69be4f72a1478ef08b for the removed tiles to not crash in certain other cases.

Comment (Mailaender) [inline OpenRA.Game/Map/TileSet.cs]: Updated.

**Comment by pchote** [comment]
Looks fine :+1:.  These tiles have been pointing to black placeholders for ages and we haven't heard of any complaints, so I doubt they are in use anymore.

File: mods/d2k/tilesets/arrakis.yaml [hunk 2]
```diff
@@ -1120,45 +1066,6 @@ Templates:
...
-		Frames: 1,1,1,1
-		Size: 2,2
-		Category: Rock-Detail
-		PickAny: False
-		Tiles:
-			0: Clear
-			1: Clear
-			3: Clear
-			2: Clear
-	Template@98: # TODO: This should go away
```
Comment (penev92) [inline mods/d2k/tilesets/arrakis.yaml]: I assume you removed this because of the comment, but it is a valid tile, so I guess the comment is wrong.

**Comment by penev92** [comment]
Apart from that one comment, the other deletions seem OK. I can't say if that covers all junk tiles, but if there are some left they can be removed when they are found.
:+1: when Template 98 is back.

File: mods/d2k/tilesets/arrakis.yaml (hunk 2 above)
Comment (Mailaender) [inline mods/d2k/tilesets/arrakis.yaml]: Fixed.

**Comment by Phrohdoh** [comment]
Played 4 different maps (in debug mode for faster testing) and couldn't spot anything broken. :+1:

**Pull request merged**